### PR TITLE
emulator for Phase-2 GCT calo jets and taus

### PR DESCRIPTION
**Pull request by pallabidas**
#### PR description:
reviewed in https://github.com/cms-l1t-offline/cmssw/pull/1161
- adds a data format: l1tp2::Phase2L1CaloJetCollection
- depends on PR https://github.com/cms-l1t-offline/cmssw/pull/1069 for input GCT towers collection (ECAL clusters + HCAL towers)
- refer to presentation here: https://indico.cern.ch/event/1288569/timetable/?view=standard#68-calorimeter-trigger-firmwar
- not a backport

### CODE REVIEW

**Comment by triggerDoctor** [comment]
Hello, I'm triggerDoctor. @aloeliger is testing this script for L1T offline software validation.

Attempts to compile this PR succeeded!

|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`eval `scramv1 runtime -sh` && scram b -j 8`|

**Comment by triggerDoctor** [comment]
Hello, I'm triggerDoctor. @aloeliger is testing this script for L1T offline software validation.

I found no issues with the code checks!
|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`eval `scramv1 runtime -sh` && scram b -k -j 8 code-checks && scram b -k -j 8 code-checks`|

I found no issues with the headers!
|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`eval `scramv1 runtime -sh` && scram b -k -j 8 check-headers`|

**Comment by triggerDoctor** [comment]
Hello, I'm triggerDoctor. @aloeliger is testing this script for L1T offline software validation.

I found 1 files that did not meet formatting requirements:
 - L1Trigger/L1CaloTrigger/plugins/L1EGammaCrystalsEmulatorProducer.cc

Please run `scram b code-format` to auto-apply code formatting
|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`eval `scramv1 runtime -sh` && scram b -k -j 8 code-format-all`|

**Comment by triggerDoctor** [comment]
Hello, I'm triggerDoctor. @aloeliger is testing this script for L1T offline software validation.

Attempts to compile this PR succeeded!

|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`eval `scramv1 runtime -sh` && scram b -j 8`|

**Comment by triggerDoctor** [comment]
Hello, I'm triggerDoctor. @aloeliger is testing this script for L1T offline software validation.

I found no issues with the code checks!
|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`eval `scramv1 runtime -sh` && scram b -k -j 8 code-checks && scram b -k -j 8 code-checks`|

I found no issues with the headers!
|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`eval `scramv1 runtime -sh` && scram b -k -j 8 check-headers`|

**Comment by triggerDoctor** [comment]
Hello, I'm triggerDoctor. @aloeliger is testing this script for L1T offline software validation.

I found no files with code format issues!
|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`eval `scramv1 runtime -sh` && scram b -k -j 8 code-format-all`|

File: L1Trigger/L1CaloTrigger/plugins/L1EGammaCrystalsEmulatorProducer.cc [hunk 1]
```diff
@@ -595,28 +602,35 @@ void L1EGCrystalClusterEmulatorProducer::produce(edm::Event& iEvent, const edm::
               }
               if (abs(hit.dieta(centerhit)) <= 2 && abs(hit.diphi(centerhit)) <= 2) {
                 e5x5 += hit.energy();
+                n5x5++;
```
Comment (aloeliger) [inline L1Trigger/L1CaloTrigger/plugins/L1EGammaCrystalsEmulatorProducer.cc]: What are these variables you're adding used for?

Comment (pallabidas) [reply L1Trigger/L1CaloTrigger/plugins/L1EGammaCrystalsEmulatorProducer.cc]: These are originally from https://github.com/cms-l1t-offline/cmssw/pull/1069

Comment (aloeliger) [reply L1Trigger/L1CaloTrigger/plugins/L1EGammaCrystalsEmulatorProducer.cc]: Okay. Then I think I'm done here

**Comment by aloeliger** [comment]
@epalencia I think we can wrap this one up too. Any comments from your side?

**Comment by epalencia** [comment]
No, looks fine, if needed we can already merged it. Otherwise, we can wait until it is merged in master. @pallabidas , could you make the corresponding PR in CMSSW?

**Comment by pallabidas** [comment]
thanks! I will make the PR to CMSSW

**Comment by triggerDoctor** [comment]
Hello, I'm triggerDoctor. @aloeliger is testing this script for L1T offline software validation.

Attempts to compile this PR succeeded!

|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`eval `scramv1 runtime -sh` && scram b -j 8`|

**Comment by triggerDoctor** [comment]
Hello, I'm triggerDoctor. @aloeliger is testing this script for L1T offline software validation.

I found no issues with the code checks!
|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`eval `scramv1 runtime -sh` && scram b -k -j 8 code-checks && scram b -k -j 8 code-checks`|

I found no issues with the headers!
|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`eval `scramv1 runtime -sh` && scram b -k -j 8 check-headers`|

**Comment by triggerDoctor** [comment]
Hello, I'm triggerDoctor. @aloeliger is testing this script for L1T offline software validation.

I found 1 files that did not meet formatting requirements:
 - L1Trigger/L1TTrackMatch/plugins/L1TrackJetClustering.h

Please run `scram b code-format` to auto-apply code formatting
|    Info   |     Value    |
|:---------:|:------------:|
|return code|`0`|
|command|`eval `scramv1 runtime -sh` && scram b -k -j 8 code-format-all`|

**Comment by epalencia** [comment]
Tagged as `phase2-l1t-1330pre3_v10`.

**Pull request merged**